src/app/devices/deviceContent/components/digitalTwin/digitalTwinInterfaces.tsx

```diff
@@ -36,18 +36,22 @@ export const DigitalTwinInterfaces: React.FC<DigitalTwinInterfacesProps> = props
     const deviceId = getDeviceIdFromQueryString(props);
     const { t } = useLocalizationContext();
 
-    const navLinks: JSX.Element[] = [];
-    if (props.idToNameMap && props.idToNameMap.size > 0) {
-        props.idToNameMap.forEach((componentName, interfaceName) => {
-            const link = `${url}${ROUTE_PARTS.DIGITAL_TWINS_DETAIL}/${ROUTE_PARTS.INTERFACES}/?${ROUTE_PARAMS.DEVICE_ID}=${encodeURIComponent(deviceId)}&${ROUTE_PARAMS.INTERFACE_ID}=${interfaceName}`;
-            navLinks.push(
+    let navLinks: JSX.Element[] = [];
+    if (props.nameToIds) {
+        navLinks = Object.keys(props.nameToIds).map(componentName => {
+            const interfaceId = (props.nameToIds as any)[componentName]; // tslint:disable-line:no-any
```

Comment:
This usage makes me believe that nameToIds is still a `Map<string, string>`

Reply:
nameToIds is the data sctructure returned by service, so I am accessing it as the object. Map value be retrieved by nameToIds.get('key'). The 'any' here gets rid of the index linter warning here.
It simply looks like `{'name1`: 'id1', 'name2': 'id2'}`

Reply:
It totally reminded me that this page has no unit test. Going to add one. Also just talked to Eddie, he suggested putting the interfaces and components into a table. I will do so in the update.